modules/aws/master/master-elb.tf

```diff
@@ -24,18 +24,44 @@ resource "aws_elb" "master" {
   tags = "${merge(
     local.common_tags,
     map(
-      "Name", "${var.cluster_name}-master"
+      "Name", "${var.cluster_name}-master-api"
     )
   )}"
 }
 
-resource "aws_elb_attachment" "master" {
-  elb      = "${aws_elb.master.id}"
-  instance = "${aws_instance.master.id}"
+resource "aws_elb" "master_etcd" {
```

Comment:
Definitely, I'm missing here some previous discussion about the architecture, but why etcd must be load balanced and reachable from the external?

Reply:
As you can see few lines below your comment there is `  internal = true` so this is internal loadblancer so its not reachable from external sources

Reply:
Sorry. Right. I missed that. So it is internal so not reachable. Another question is: for which use cases it is used?

Reply:
calico is connecting to etcd so its mostly for calico

Reply:
Right calico need etcd endpoints in the configuration. This might be another proposal for SIG Infra. What about to remove the ELB from the equation in the future and pass to calico the DNS name of each master to the etcd port (2379 if I'm not wrong). of course, this introduces the problem to keep update the DNS record with the IP address of the master. But this should be probably easily solvable with the introduction of a bit of automation which come with a tool like external-dns. would be possible or do you find any big constraints? The reason for this proposal is because I'm always happy when I'm able to remove from the equation piece of external infra like ELB which participate to increase the bill and have a more neat infra.

Reply:
That might work too, lets create issue fro that and we can easily adress later.

Reply:
Going to do that ;-)